sd_dynamic_prompts/ui/pnginfo_saver.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class ImagePrompts:
+    prompt: str
+    negative_prompt: str
+
+
+class PngInfoSaver:
+    def __init__(self):
+        self._enabled = True
+
+    def set_enabled(self, enabled: bool) -> None:
+        self._enabled = enabled
+
+    def update_pnginfo(self, parameters: str, image_prompts: ImagePrompts) -> str:
+        if not self._enabled:
+            return parameters
+
+        if image_prompts.prompt != "":
+            parameters += "\nTemplate:" + image_prompts.prompt
+
+        if image_prompts.negative_prompt != "":
+            parameters += "\nNegative Template:" + image_prompts.negative_prompt
```

Comment:
* No need for `!= ""`; empty strings are falsy
* f-strings are nicer than concatenation
* a space between `template:` and the prompt? :)

```suggestion
        if image_prompts.prompt:
            parameters += f"\nTemplate: {image_prompts.prompt}"

        if image_prompts.negative_prompt:
            parameters += f"\nNegative Template: {image_prompts.negative_prompt}"
```

Reply:
happy with that - but I've never liked truthy and falsy since they aren't very explicit but I should probably just get over it :)